src/randomInt.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Generate an inclusive random integer between `from` and `to`.
+ *
+ * @param from - The minimum value.
+ * @param to - The maximum value.
+ * @returns The random integer.
+ * @signature
+ *   R.randomInt(from, to)
+ * @example
+ *   R.randomInt(1, 10) // => 5
+ *   R.randomInt(1n, 10n) // => 7n
+ * @dataFirst
+ * @category Number
+ */
+export function randomInt<T extends bigint | number>(
+  from: T,
+  to: Widen<T>,
+): Widen<T> {
+  if (to <= from) {
+    throw new RangeError(`to(${to}) should be greater than from(${from})`);
+  }
+
+  if (typeof from === "bigint" && typeof to === "bigint") {
+    return BigInt(
+      Math.floor(Number(to - from + BigInt(1)) * Math.random() + Number(from)),
+    ) as never;
+  }
+
+  if (!Number.isFinite(from)) {
+    throw new TypeError(`from(${from}) is not a finite number`);
+  }
+
+  if (!Number.isFinite(to)) {
+    throw new TypeError(`to(${to}) is not a finite number`);
+  }
+
+  // Cast the values to numbers since TypeScript can't infer it properly.
+  const _from = from as number;
+  const _to = to as number;
+
+  if (Math.floor(_from) === Math.floor(_to)) {
+    throw new RangeError(
+      `there are no integers between from(${from}) and to(${to})`,
+    );
+  }
```

Comment:
Why does this matter? if someone has some code that fetches two numbers (maybe a UI) and it results in calling `randomInt(2, 2)`, why shouldn't we just return `2` for them consistently, throwing here might break people's code.

We prefer not throwing unless the algorithm cannot work without preemptively throwing.

```suggestion
```

Reply:
it was mostly for cases like `randomInt(1.5, 1.6)`
what should be returned in this case?
the MDN implementation returns 2 (which is not what I would've expected TBH, I think it should _probably_ be 1)

Reply:
If the result of `Math.ceil(from)` is smaller than the result of `Math.floor(to)` you should throw a RangeError, and shouldn't throw an error for any other case.